Ticket-booking-spring-app/src/main/java/com/example/demo/service/reservation/ReservationServiceImpl.java

```diff
@@ -0,0 +1,181 @@
+package com.example.demo.service.reservation;
+
+import com.example.demo.dao.ReservationRepository;
+import com.example.demo.entity.*;
+import com.example.demo.exception.DataBaseException;
+import com.example.demo.exception.InvalidReservationException;
+import com.example.demo.service.reservation_detail.ReservationDetailService;
+import com.example.demo.service.room.RoomService;
+import com.example.demo.service.screening.ScreeningService;
+import com.example.demo.service.seat.SeatService;
+import com.example.demo.service.seat_availability.SeatAvailabilityService;
+import com.example.demo.service.ticket_price.TicketPriceService;
+import com.example.demo.util.reservation_request.ReservationRequest;
+import com.example.demo.util.RoomLayout;
+import com.example.demo.util.reservation_request.TicketRequest;
+import com.example.demo.util.reservation_request.TicketType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class ReservationServiceImpl implements ReservationService {
+    private static int expirationMinutesBeforeScreening = 10;
+    private ReservationRepository reservationRepository;
+    private ScreeningService screeningService;
+    private SeatAvailabilityService seatAvailabilityService;
+    private SeatService seatService;
+    private TicketPriceService ticketPriceService;
+    private RoomService roomService;
+    private ReservationDetailService reservationDetailService;
+
+    @Autowired
+    public ReservationServiceImpl(
+            ReservationRepository theReservationRepository,
+            ScreeningService theScreeningService,
+            SeatAvailabilityService theSeatAvailabilityService,
+            TicketPriceService theTicketPriceService,
+            SeatService theSeatService,
+            RoomService theRoomService,
+            ReservationDetailService theReservationDetailService) {
+        reservationRepository = theReservationRepository;
+        screeningService = theScreeningService;
+        seatAvailabilityService = theSeatAvailabilityService;
+        ticketPriceService = theTicketPriceService;
+        seatService = theSeatService;
+        roomService = theRoomService;
+        reservationDetailService = theReservationDetailService;
+    }
+
+    @Override
+    public Reservation postReservation(ReservationRequest reservationRequest) {
+        List<Seat> wantedSeats = getRequestedSeats(reservationRequest);
+        Reservation reservation = reservationRepository.
+                save(processReservationRequest(reservationRequest));
+
+        for (Seat seat : wantedSeats) {
```

Comment:
Niby w pomie zadeklarowana java 17, a tutaj wygląda jakby była tylko Java 5 :(

Reply:
Co tutaj można zrobić? Jedyne co mi przychodzi do głowy to 'var' albo stream().forEach, ale to jakoś szczególnie nie upraszcza tego kodu.